src/Middleware/Diagnostics/src/DeveloperExceptionPage/Views/ErrorPage.cshtml

```diff
@@ -21,312 +21,341 @@
-                        }
-                    </tbody>
-                </table>
-            }
-            else
-            {
-                <p>@Resources.ErrorPageHtml_NoEndpoint</p>
-            }
-            <h2>@Resources.ErrorPageHtml_RouteValues</h2>
-            @if (Model.RouteValues != null && Model.RouteValues.Any())
-            {
-                <table>
-                    <thead>
+                    }
+                </tbody>
+            </table>
+        }
+        else
+        {
+            <p>@Resources.ErrorPageHtml_NoEndpoint</p>
+        }
+        <h2>@Resources.ErrorPageHtml_RouteValues</h2>
+        @if (Model.RouteValues != null && Model.RouteValues.Any())
+        {
+            <table>
+                <thead>
+                    <tr>
+                        <th>@Resources.ErrorPageHtml_VariableColumn</th>
+                        <th>@Resources.ErrorPageHtml_ValueColumn</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach (var kv in Model.RouteValues.OrderBy(kv => kv.Key))
+                    {
                         <tr>
-                            <th>@Resources.ErrorPageHtml_VariableColumn</th>
-                            <th>@Resources.ErrorPageHtml_ValueColumn</th>
+                            <td>@kv.Key</td>
+                            <td>@(kv.Value!)</td>
                         </tr>
-                    </thead>
-                    <tbody>
-                        @foreach (var kv in Model.RouteValues.OrderBy(kv => kv.Key))
-                        {
-                            <tr>
-                                <td>@kv.Key</td>
-                                <td>@(kv.Value!)</td>
-                            </tr>
-                        }
-                    </tbody>
-                </table>
-            }
-            else
-            {
-                <p>@Resources.ErrorPageHtml_NoRouteValues</p>
-            }
-        </div>
-
-        <script>
-            //<!--
-            <%$ include: ErrorPage.js %>
-            //-->
-        </script>
-    </body>
+                    }
+                </tbody>
+            </table>
+        }
+        else
+        {
+            <p>@Resources.ErrorPageHtml_NoRouteValues</p>
+        }
+        <h2>@Resources.ErrorPageHtml_Metadata</h2>
+        @if (Model.Endpoint?.Metadata != null)
+        {
+            <table>
+                <thead>
+                    <tr>
+                        <th>@Resources.ErrorPageHtml_MetadataTypeColumn</th>
+                        <th>@Resources.ErrorPageHtml_MetadataDetail</th>
+                    </tr>
+                </thead>
+                <tbody>
+                    @foreach (var metadata in Model.Endpoint.Metadata)
+                    {
+                        <tr>
+                            <td>@metadata.GetType().Name</td>
+                            <td>@(metadata.ToString())</td>
```

Comment:
Does Razor escape HTML by default? I want to make sure that a ToString result that returns HTML doesn't accidently break the page.

Reply:
Also, what happens if metadata has new lines? They should be kept when displayed on the web page.

Reply:
I think `HtmlEncodeAndReplaceLineBreaks` (which the title above uses) should be called here.